addons/mail/models/discuss/ir_attachment.py

```diff
@@ -27,3 +27,18 @@ def _get_upload_env(self, thread_model, thread_id):
                 .env
             )
         return super()._get_upload_env(thread_model, thread_id)
+
+    def _prepare_attachment_format(self, attachment):
+        attachment_format = super()._prepare_attachment_format(attachment)
+        voice = self.env["discuss.voice.metadata"].sudo().search([("attachment_id", "=", attachment.id)])
```

Comment:
this should be done in batch, not having one search per attachment (query count should not be increased as much)

Reply:
note that defining the inverse of `discuss.voice.metadata` `attachment_id` on attachment and using it might help take care of prefetch and cache to reduce queries too